Minor fixes

ypid opened the pull request.

ypid committed:
[debops-contrib.tor] Change inventory file prefix from none to to .yml

ypid committed:
[debops-contrib.x2go_server] Change inventory file prefix from to .yml

ypid committed:
[etckeeper] Ignore etesync-server/secret.txt

ypid committed:
[netbox] No need to use `&&` with `set -o errexit`

ypid committed:
[owncloud] Update main supported distro to buster

ypid committed:
[docs] Move volkszaehler to end user applications in role index

Volkszaehler is similar to Kibana in that both are able to show
graphs/diagrams to end users.

ypid committed:
[docs] Fix typos

ypid committed:
[python] Expose Python version in Ansible local facts

ypid committed:
[resources] File deletion should be done before creation

ypid committed:
[resources] Add Ansible tag role::resources:commands

ypid committed:
[python] Fix coding style: E501 line too long (171 > 79 characters)

drybjed commented on ansible/roles/etckeeper/defaults/main.yml:
```diff
@@ -176,6 +176,9 @@ etckeeper__default_gitignore:
     ignore: 'zfs/zpool.cache'
 
+  - name: 'etesync-secret'
+    ignore: 'etesync-server/secret.txt'
```
That's a bit random, isn't it? It's the first time I hear about [EteSync](https://www.etesync.com), it's not even available in Debian. I don't see an RFP or ITP messages about it either. There's no efforts to create a `.deb` package in its GitHub repositories that I could see on a quick glance, looking for an Ansible playbook/role for it also doesn't show any results, nothing on Ansible Galaxy...

The `etckeeper` role is designed to handle such cases via Ansible inventory perfectly fine, so why add this particular case to the role defaults?

ypid replied:
I guess because I have a DebOps role 80 % done for a EteSync server and adding it to etckeeper ignores appeared as one usual preparation work for me :)

drybjed replied:
Very well then, it can stay. :-)

drybjed commented on ansible/roles/python/templates/etc/ansible/facts.d/python.fact.j2:
```diff
@@ -29,4 +31,29 @@ output = loads("""{{ ({'enabled':  python__enabled | bool,
+        ])
+    except subprocess.CalledProcessError:
+        pass
+
```
It's a good idea to add the available Python version to the facts, however it's kind of redundand since Ansible does this in its "internal" facts as well, although only for the interpreter. However, why did you go with such an elaborate way to do this? Why not just go with parsing the output of `python3 --version` and `python2 --version` instead? I'm curious because this script will be executed frequently during Ansible runs, so I'm not sure which method would be faster.

ypid replied:
Because `platform.python_version` is the API driven way to do it and I expect it to be more stable/reliable. I did some testing and it does not make much of a difference but I changed it anyway.

One of those version checks could be switched to Ansible "internal" facts. But not both. For my use case, it did not work obviously thus I added it to the local facts. I still have Python2 as `interpreter_python` but I want to run EteSync (for which this Python version check is for) under Python3.

drybjed reviewed (commented):
Thanks for your improvements! However there are some questions I would like you to answer first. See the comments below.

ypid committed:
[python] Switch to `python3 --version` calls instead of platform module

Because it is faster:

$ time python3 --version
Python 3.7.3
python3 --version  0.01s user 0.00s system 90% cpu 0.010 total

$ time python3 -c "import platform; print(platform.python_version())"
3.7.3
0.03s user 0.01s system 98% cpu 0.041 total

No proper benchmarking was done but you can kinda guess why `python3 --version` is faster:

* Does not have to interpret (Python) code.
* Only executes the python binary and can exit right after the command
  line args parsing.

ypid committed:
[python] Make local facts resilient to absence of python2
